app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt

```diff
@@ -79,14 +84,15 @@ class KiwixMainActivity : CoreMainActivity() {
     bottom_nav_view.setupWithNavController(navController)
   }
 
-  override fun onPostCreate(savedInstanceState: Bundle?) {
-    super.onPostCreate(savedInstanceState)
+  override fun onStart() {
+    super.onStart()
     navController.addOnDestinationChangedListener { _, destination, _ ->
       bottom_nav_view.isVisible = destination.id in topLevelDestinations
       if (destination.id !in topLevelDestinations) {
         handleDrawerOnNavigation()
       }
     }
+    navigate(initialDestinationFragmentId)
```

Comment:
So if I press the home button and then return to the app through the recents screen we will navigate to the "initialDestination" https://developer.android.com/reference/android/app/Activity#ActivityLifecycle

This needs to be rethought

Reply:
Yeah, I don't know why I made it so complicated when we can just check in onStart if we need to start intro 🤦‍♂️